nntrainer/src/conv2d_layer.cpp

```diff
@@ -176,9 +174,17 @@ sharedConstTensor Conv2DLayer::backwarding(sharedConstTensor derivative,
       delBias.setValue(0, 0, 0, 0, sum + delBias.getValue(0, 0, 0, 0));
     }
   }
+  return;
+}
 
-  // Calculate delS : returns ( Full pad )
+sharedConstTensor Conv2DLayer::backwarding(sharedConstTensor derivative,
+                                           int iteration) {
+
+  // Calculate delK : [batch, channel, height, width ] * filter_size
+  std::thread delk_thread(&Conv2DLayer::calculateDelK, this, derivative);
```

Comment:
Lets not create new thread everytime, but create 1 thread at runtime and reuse it existing thread.